src/compiler/program.ts

```diff
@@ -2200,6 +2200,12 @@ namespace ts {
                 }
             }
 
+            if (options.resolveJsonModule) {
+                if (getEmitModuleResolutionKind(options) !== ModuleResolutionKind.NodeJs) {

```

Comment:
what about `--module`? do not think we should allow this for `--module ESNext` for example..

Reply:
Clarified offline. At present ```--moduleResolution``` specified takes preferences for resolution irrespective of ```--module``` and since there is no change in there, no error needs to be reported specially.